others/longest_substring_without_repeating_characters.cpp

```diff
@@ -0,0 +1,109 @@
+/**
+ * @file
+ * @brief Solution for Longest Substring Without Repeating Characters problem.
+ * @details
+ * Problem link: https://leetcode.com/problems/longest-substring-without-repeating-characters/description/
+ * 
+ * Intuition:
+ * 1) The intuition is straightforward and simple. We track the frequency of characters.
+ * 2) Since we can't use a string to track the longest substring without repeating characters efficiently (as removing a character from the front of a string isn't O(1)), we optimize the solution using a deque approach.
+ *
+ * Approach:
+ * 1) Initialize an unordered_map to track the frequency of characters.
+ * 2) Use a deque for pushing characters, and update the result deque (`res`) with the current deque (`temp`) 
+ *    whenever we find a longer substring.
+ * 3) Use a while loop to reduce the frequency from the front, incrementing `i`, 
+ *    and removing characters from the `temp` deque as we no longer need them.
+ * 4) Return `res.size()` as we are interested in the length of the longest substring.
+ * 
+ * Time Complexity: O(N)
+ * Space Complexity: O(N)
+ * 
+ * I hope this helps to understand.
+ * Thank you!
+ **/
+
+#include <iostream>        // for input and output read/write.
+#include <unordered_map>   // for character frequency map.
+#include <deque>           // for push and pop operations at O(1) time.
+#include <string>          // for taking string as input.
+#include <cassert>         // for assert.
```

Comment:
These should mention what functions/types are used from the pr

Reply:
* Have done the required changes why the header file is used for which function.
* Thank you.